client/planning-extension/src/assignments-overview/index.tsx

```diff
@@ -0,0 +1,245 @@
+            acc[item._id] = item;
+
+            return acc;
+        }, {});
+
+        return desks;
+    });
+}
+
+function fetchContentTypes(): Promise<IState['contentTypes']> {
+    return findOne<IVocabulary>('vocabularies', 'g2_content_type').then(({items}) => (items));
+}
+
+function fetchAssignments(userId: IUser['_id']): Promise<IState['assignments']> {
+    return queryRawJson<IRestApiResponse<IAssignmentItem>>(
+        'assignments',
+        {
+            page: '1',
+            sort: '[("planning.scheduled", 1)]',
+            source: JSON.stringify({
+                query: {
+                    bool: {
+                        must: [
+                            {term: {'assigned_to.user': userId}},
+                            {terms: {'assigned_to.state': ['assigned', 'submitted', 'in_progress']}},
+                        ],
+                    },
+                },
+            }),
+        },
+    ).then(({_items}) => {
+        superdesk.dispatchEvent(
+            'menuItemBadgeValueChange',
+            {menuId: 'MENU_ITEM_PLANNING_ASSIGNMENTS', badgeValue: _items.length.toString()},
+        );
+
+        return _items;
+    });
+}
+
+export class AssignmentsList extends React.PureComponent<{}, {loading: true} | IState> {
+    private eventListenersToRemoveBeforeUnmounting: Array<() => void>;
+
+    constructor(props: {}) {
+        super(props);
+
+        this.state = {loading: true};
+
+        this.eventListenersToRemoveBeforeUnmounting = [];
+
+        this.handleContentChanges = this.handleContentChanges.bind(this);
+
+        this.eventListenersToRemoveBeforeUnmounting.push(
+            addWebsocketMessageListener(
+                'resource:created',
+                (event) => {
+                    const {resource, _id} = event.detail.extra;
+
+                    this.handleContentChanges([{changeType: 'created', resource: resource, itemId: _id}]);
+                },
+            ),
+        );
+
+        this.eventListenersToRemoveBeforeUnmounting.push(
+            addWebsocketMessageListener(
+                'resource:updated',
+                (event) => {
+                    const {resource, _id, fields} = event.detail.extra;
+
+                    this.handleContentChanges([{
+                        changeType: 'updated',
+                        resource: resource,
+                        itemId: _id,
+                        fields: fields,
+                    }]);
+                },
+            ),
+        );
+
+        this.eventListenersToRemoveBeforeUnmounting.push(
+            addWebsocketMessageListener(
+                'resource:deleted',
+                (event) => {
+                    const {resource, _id} = event.detail.extra;
+
+                    this.handleContentChanges([{changeType: 'deleted', resource: resource, itemId: _id}]);
+                },
+            ),
+        );
+    }
+
+    handleContentChanges(changes: Array<IResourceChange>) {
```

Comment:
not sure if I'm missing something but seems like it's always just single change, so why there is array?